lib/screens/calendar/calendar.dart

```diff
@@ -0,0 +1,841 @@
+                                      HighLightSingle(),
+                                      CenterText(Colors.white, day),
+                                    ],
+                                  );
+                                } else {
+                                  return Stack(
+                                    children: [
+                                      HighLightOpen(),
+                                      CenterText(Colors.white, day),
+                                    ],
+                                  );
+                                }
+                              } else if (!eventsPeriod.contains(
+                                  convertToIso8601inDays(
+                                      day.add(Duration(days: 1))))) {
+                                return Stack(
+                                  children: [
+                                    HighLightClose(),
+                                    CenterText(Colors.white, day),
+                                  ],
+                                );
+                              } else {
+                                return Stack(
+                                  children: [
+                                    HighLightMiddle(),
+                                    CenterText(Colors.white, day),
+                                  ],
+                                );
+                              }
+                            } else
+                              return CenterText(Colors.white, day);
+                          },
+                          outsideBuilder: (context, day, focusedDay) {
+                            var dayString = convertToIso8601inDays(day);
+                            var thatDayinHLIndex =
+                                eventsPeriod.indexOf(dayString);
+                            var previousDayinHLIndex =
+                                eventsPeriod.indexOf(dayString) - 1;
+                            var afterDayinHLIndex =
+                                eventsPeriod.indexOf(dayString) + 1;
+                            var dayS = day.toString();
+
+                            return Stack(
+                              children: <Widget>[
+                                Center(
+                                  child: Container(
+                                    width: 30,
+                                    height: 30,
+                                    child: CenterText(Colors.grey, day),
+                                  ),
+                                ),
+                                eventsPeriod
+                                        .contains(convertToIso8601inDays(day))
+                                    ? (thatDayinHLIndex == 0 ||
+                                            previousDayinHLIndex.isNegative ||
+                                            !eventsPeriod.contains(
+                                                convertToIso8601inDays(
```

Comment:
there is a lot of conditions going on here. Can you simplify all of these with variables?